cmd/ci-secret-bootstrap/main.go

```diff
@@ -212,24 +216,40 @@ func constructSecrets(config []secretConfig, bwClient bitwarden.Client) (map[str
 	return secretsMap, nil
 }
 
-func updateSecrets(secretsGetters map[string]coreclientset.SecretsGetter, secretsMap map[string][]*coreapi.Secret) error {
+func updateSecrets(secretsGetters map[string]coreclientset.SecretsGetter, secretsMap map[string][]*coreapi.Secret, force bool) error {
 	for cluster, secrets := range secretsMap {
 		for _, secret := range secrets {
 			logrus.Infof("handling secret: %s:%s/%s", cluster, secret.Namespace, secret.Name)
 			secretsGetter := secretsGetters[cluster]
-			if _, err := secretsGetter.Secrets(secret.Namespace).Create(secret); err != nil {
-				if !kerrors.IsAlreadyExists(err) {
-					return err
-				}
-				if _, err := secretsGetter.Secrets(secret.Namespace).Update(secret); err != nil {
+			existingSecret, err := secretsGetter.Secrets(secret.Namespace).Get(secret.Name, meta.GetOptions{})
+			if kerrors.IsNotFound(err) {
+				if _, err := secretsGetter.Secrets(secret.Namespace).Create(secret); err != nil {
 					return err
 				}
+			} else if err != nil {
+				return err
+			}
+			if !force && !equality.Semantic.DeepEqual(secret.Data, existingSecret.Data) {
```

Comment:
Lets do that right after we Get the secret.
 And we want to update only if that passes right?

It seems that you need something like:

```go
			if existingSecret, err := secretsGetter.Secrets(secret.Namespace).Get(secret.Name, meta.GetOptions{}); err == nil {
				if !force && !equality.Semantic.DeepEqual(secret.Data, existingSecret.Data) {
					// logrus....
					return fmt.Errorf("found unsynchronized secret: %s:%s/%s", cluster, secret.Namespace, secret.Name)
				}

				if _, err := secretsGetter.Secrets(secret.Namespace).Update(secret); err != nil {
					return err
				}

			} else if kerrors.IsNotFound(err) {
				if _, err := secretsGetter.Secrets(secret.Namespace).Create(secret); err != nil {
					return err
				}
			} else
				return err
			}
```

Reply:
Good catch. Updated.